### PR TITLE
 feat(Pagination): initial

### PR DESCRIPTION
**Pull request by zouxuoz**

### CODE REVIEW

File: src/components/Pagination/Pagination.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+          <Select
+            className={ css`width: 64px; margin-left: 12px;` }
+            value={ pageSize }
+            options={ this.getPageSizeOptions(pageSizeOptions) }
+            onChange={ this.onChangePageSize }
+          />
+        </If>
+
+        <Text className={ css`margin-left: 12px;` }>
+          { numeral((page - 1) * pageSize + 1).format('0,0') } - { numeral(Math.min(total, page * pageSize)).format('0,0') } of { numeral(total).format('0,0') } records
```
Comment (vorobeez) [inline src/components/Pagination/Pagination.js]: Why don't you use `Int.NumberFormat`?

This formatter does the same:
```
const formatter = new Intl.NumberFormat('en-EN', { useGrouping: true, maximumFractionDigits: 0 })
```

File: src/components/Pagination/Pagination.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+  // $FlowFixMe
+  onChange = memoize((page) => () => {
+    const { onChange, pageSize: pageSizeFormProps } = this.props;
+
+    this.setState(({ pageSize: pageSizeFromState }) => {
+      const pageSize = pageSizeFormProps || pageSizeFromState;
+
+      if (typeof onChange === 'function') {
+        onChange(page, pageSize);
```
Comment (vorobeez) [inline src/components/Pagination/Pagination.js]: Is it fine to call `onChange` inside of `setState` function?

File: src/components/Pagination/Pagination.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    super(props);
+
+    this.state = {
+      page: props.defaultPage,
+      pageSize: props.defaultPageSize,
+    };
+  }
+
+  // $FlowFixMe
+  onChange = memoize((page) => () => {
```
Comment (vorobeez) [inline src/components/Pagination/Pagination.js]: Why do you use `memoize` here?

Comment (zouxuoz) [reply src/components/Pagination/Pagination.js]: @vorobeez for rid of recreating `onChange` handler for each button on each render, but I think I can rename it for more readability

File: src/components/Pagination/Pagination.js (hunk 2 above)
Comment (zouxuoz) [reply src/components/Pagination/Pagination.js]: @vorobeez  fIxed

File: src/components/Pagination/Pagination.js (hunk 1 above)
Comment (zouxuoz) [reply src/components/Pagination/Pagination.js]: @vorobeez good idea

File: src/components/Pagination/Pagination.js (hunk 3 above)
Comment (vorobeez) [reply src/components/Pagination/Pagination.js]: Oh, i didn't catch nested function.

**Pull request merged**